src/lib/dirsize.c

```diff
@@ -130,9 +131,12 @@ double libreport_get_dirsize_find_largest_dir(
                     }
                     else
                     {
-                        maxsz = sz;
-                        free(*worst_dir);
-                        *worst_dir = libreport_xstrdup(ep->d_name);
+                        if (strcmp(ep->d_name, proc_dir))
```

Comment:
This segfaults if `proc_dir == NULL`.

Also, I'd prefer to have the explicit `!= 0` there.